src/secp.rs

```diff
@@ -0,0 +1,194 @@
+pub use secp256k1zkp::{ContextFlag, Message, Secp256k1, Signature};
+pub use secp256k1zkp::ecdh::SharedSecret;
+pub use secp256k1zkp::pedersen::{Commitment, RangeProof};
+pub use secp256k1zkp::key::{PublicKey, SecretKey};
+pub use secp256k1zkp::constants::{AGG_SIGNATURE_SIZE, COMPRESSED_PUBLIC_KEY_SIZE, MAX_PROOF_SIZE, PEDERSEN_COMMITMENT_SIZE, SECRET_KEY_SIZE};
+
+use crate::ser::{Readable, Reader, Writeable, Writer};
+use crate::error::{ErrorKind, Result};
+
+use rand::RngCore;
+use std::cmp;
+
+/// A generalized Schnorr signature with a pedersen commitment value & blinding factors as the keys
+pub const COM_SIGNATURE_SIZE : usize = 96;
+
+pub struct ComSignature(pub [u8; COM_SIGNATURE_SIZE]);
+impl ComSignature {
+    /// Builds a ComSignature from a byte vector. If the vector is too short, it will be
+    /// completed by zeroes. If it's too long, it will be truncated.
+    pub fn from_vec(v: Vec<u8>) -> ComSignature {
+        let mut h = [0; COM_SIGNATURE_SIZE];
+        for i in 0..cmp::min(v.len(), COM_SIGNATURE_SIZE) {
+            h[i] = v[i];
+        }
+        ComSignature(h)
+    }
+
+    #[allow(dead_code)]
+    pub fn sign(_value: u64, _blind: &SecretKey, _msg: &Vec<u8>) -> Result<ComSignature> {
+        // milestone 2 - todo
+        let mut h = [0u8; COM_SIGNATURE_SIZE];
+        for i in 0..COM_SIGNATURE_SIZE {
+            h[i] = i as u8;
+        }
+        Ok(ComSignature(h))
+    }
+
+    pub fn verify(self, _commit: &Commitment, _msg: &Vec<u8>) -> Result<()> {
+        // milestone 2 - todo
+        Ok(())
+    }
+}
+
+impl AsRef<[u8]> for ComSignature {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+/// Serializes a ComSignature to and from hex

```

Comment:
Not sure if really needed, but when the comsig is implemented, we could write a simple test that asserts the validity of the signature is preserved e.g. `is_valid(deser(ser(sig), pubkey, msg)` still holds.